trlx/trainer/accelerate_base_trainer.py

```diff
@@ -77,6 +77,12 @@ def __init__(self, config, **kwargs):  # noqa: C901
             self.tokenizer.pad_token = self.tokenizer.eos_token
             self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
 
+        if self.additional_special_tokens is not None and type(self.additional_special_tokens) is list:
+            self.tokenizer.add_special_tokens(
+                {"additional_special_tokens": self.additional_special_tokens}
+            )
+            self.model.base_model.resize_token_embeddings(len(self.tokenizer))
```

Comment:
To improve compatibility with other modified tokenizers, I think it would be great if resizing happened by default, regardless of this if condition. Also, for PPO, the reference model/head should be resized likewise, otherwise, this error occurs:
```
../aten/src/ATen/native/cuda/ScatterGatherKernel.cu:144: operator(): block: [0,0,0], thread: [93,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
Traceback (most recent call last):
  File "/trlx/examples/ppo_sentiments.py", line 58, in <module>
    main(hparams)
  File "/trlx/examples/ppo_sentiments.py", line 47, in main
    trlx.train(
  File "/trlx/trlx/trlx.py", line 133, in train
    trainer.learn()
  File "/trlx/trlx/trainer/accelerate_base_trainer.py", line 506, in learn
    self.prepare_learning()
  File "trlx/trlx/trainer/accelerate_ppo_trainer.py", line 239, in prepare_learning
    self.make_experience(self.config.method.num_rollouts)
  File "/trlx/trlx/trainer/accelerate_ppo_trainer.py", line 427, in make_experience
    ref_logprobs = logprobs_of_labels(ref_logits[:, :-1, :], all_tokens[:, 1:])
  File "/trlx/trlx/utils/modeling.py", line 224, in logprobs_of_labels
    logprobs_labels = torch.gather(logprobs, dim=-1, index=labels.unsqueeze(-1))
RuntimeError: CUDA error: device-side assert triggered
```

Reply:
Thanks for your review, I will resolve it later. 
My plan is:
* if hydra heads is used, `hasattr(self.model, "frozen_head")`, then I need to  resize the `self.model.frozen_head.decoder_blocks`, 
* if not, just resize the `self.ref_model`